clang/lib/CodeGen/CGPointerAuthInfo.h

```diff
@@ -83,14 +85,26 @@ class CGPointerAuthInfo {
 
   friend bool operator!=(const CGPointerAuthInfo &LHS,
                          const CGPointerAuthInfo &RHS) {
-    return LHS.Key != RHS.Key || LHS.Discriminator != RHS.Discriminator ||
-           LHS.AuthenticationMode != RHS.AuthenticationMode;
+    return !(LHS == RHS);
   }
 
   friend bool operator==(const CGPointerAuthInfo &LHS,
                          const CGPointerAuthInfo &RHS) {
-    return !(LHS != RHS);
+    auto AsTuple = [](const CGPointerAuthInfo &Info) {
+      return std::make_tuple(Info.AuthenticationMode, Info.IsIsaPointer,
+                             Info.AuthenticatesNullValues, Info.Key,
+                             Info.IntDiscriminator, Info.AddrDiscriminator);
+    };
+    return AsTuple(LHS) == AsTuple(RHS);
   }
+
+private:
+  PointerAuthenticationMode AuthenticationMode : 2;
+  unsigned IsIsaPointer : 1;
+  unsigned AuthenticatesNullValues : 1;
+  unsigned Key : 2;
+  unsigned IntDiscriminator : 16;
```

Comment:
Is there any benefit from using `unsigned : 16` instead of `uint16_t` here? I guess that with the first option we do not have alignment restriction, but the `Value *` pointer below should be anyway aligned as 8 (for 64-bit systems), so probably we do not need to count every bit and we can just use `uint16_t` here.

I might be wrong, just please explain your decision.